build.rs

```diff
@@ -105,4 +105,83 @@ pub fn main() {
     mf.write_all(b"\n}\n").unwrap();
 
     mf.flush().unwrap();
+
+    // Always generate embedded English locale files
+    // This ensures English works even when locale files aren't available (e.g., cargo install)
+    generate_embedded_english_locales(&out_dir, &crates).unwrap();
+}
+
+/// Generate embedded English locale files
+///
+/// # Errors
+///
+/// Returns an error if file operations fail or if there are I/O issues
+fn generate_embedded_english_locales(
+    out_dir: &str,
+    crates: &[String],
+) -> Result<(), Box<dyn std::error::Error>> {
+    use std::fs;
+
+    let mut embedded_file = File::create(Path::new(&out_dir).join("embedded_locales.rs"))?;
+
+    writeln!(embedded_file, "// Generated at compile time - do not edit")?;
+    writeln!(
+        embedded_file,
+        "// This file contains embedded English locale files for all utilities"
+    )?;
+    writeln!(embedded_file)?;
+    writeln!(embedded_file, "use std::collections::HashMap;")?;
+    writeln!(embedded_file)?;
+
+    // Start the function that returns embedded locales
+    writeln!(
+        embedded_file,
+        "pub fn get_embedded_locales() -> HashMap<&'static str, &'static str> {{"
+    )?;
+    writeln!(embedded_file, "    let mut locales = HashMap::new();")?;
+    writeln!(embedded_file)?;
+
+    // Embed locale files for each utility
```

Comment:
To make sure I understand... This embeds the locales in individual binaries as well right? (not just the multi-call one?)

If that's the case, won't that end up blowing up individual binaries sizes? I wonder if it'd be possible to embed only the translations needed for each binary.

If not, well, we'd still need to embed in individual binaries to fix #8382, so...

(also, I'd be ok to merge this, and optimize later)

Reply:
> To make sure I understand... This embeds the locales in individual binaries as well right? (not just the multi-call one?)

oh crap, right, i need to fix that